discord-scripts/fix-linear-embed.ts

```diff
@@ -132,32 +160,60 @@ async function processLinearEmbeds(
   logger: Log,
   linearClient: LinearClient,
 ) {
-  const issueUrlRegex =
-    /https:\/\/linear\.app\/([a-zA-Z0-9-]+)\/issue\/([a-zA-Z0-9-]+)(?:.*#comment-([a-zA-Z0-9]+))?/g
+  if (!issueTagRegex) {
+    logger.error("IssueTagRegex is not initialized.")
+    return
+  }
 
-  const matches = Array.from(message.matchAll(issueUrlRegex))
+  const urlMatches = Array.from(message.matchAll(issueUrlRegex))
+  const issueMatches = Array.from(message.matchAll(issueTagRegex))
 
-  if (matches.length === 0) {
+  if (urlMatches.length === 0 && issueMatches.length === 0) {
     return
   }
 
   const processedIssues = processedMessages.get(messageId) || new Set<string>()
   processedMessages.set(messageId, processedIssues)
 
-  const embedPromises = matches.map(async (match) => {
+  const uniqueMatches = new Set<string>()
+
+  urlMatches.forEach((match) => {
     const teamName = match[1]
     const issueId = match[2]
     const commentId = match[3] || undefined
-    const uniqueKey = `${issueId}-${commentId}`
+    const uniqueKey = `${issueId}-${commentId || ""}`
 
-    if (processedIssues.has(uniqueKey)) {
-      return null
+    if (!processedIssues.has(uniqueKey)) {
+      processedIssues.add(uniqueKey)
+      uniqueMatches.add(JSON.stringify({ issueId, commentId, teamName }))
     }
+  })
+
+  issueMatches.forEach((match) => {
+    const issueId = match[0]
+
+    if (
+      Array.from(uniqueMatches).some(
+        (uniqueMatch) => JSON.parse(uniqueMatch).issueId === issueId,
```

Comment:
Why not just `uniqueMatches.has("${issueId}-")`?

Reply:
@Shadowfiend Appreciate the great review comments 🙏 . Just committed a bunch of fixes and refactoring to better handle these cases. Especially when dealing with the back-and-forth json tango here 😁